model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java

```diff
@@ -1057,7 +1083,7 @@ public Stream<GroupModel> searchGroupsByAttributes(RealmModel realm, Map<String,
 
         TypedQuery<GroupEntity> query = em.createQuery(queryBuilder);
         return closing(paginateQuery(query, firstResult, maxResults).getResultStream())
-                .map(g -> session.groups().getGroupById(realm, g.getId()));
+                .map(g -> new GroupAdapter(session, realm, em, g));
```

Comment:
Why is this changed? The purpose of calling `getGroupById` is to store the entity within the Infinispan cache.

Reply:
That makes sense! When I originally went over this code to add the pagination the reasoning for the extra query wasn't apparent, since it reads as "map the group entity to the same entity after doing another lookup" 

Because my focus for this PR was thinking about scale, I wanted to remove the extra database interaction, thinking it was an accidental duplication. 

In reality, most queries that have to worry about the overhead from the extra database call are probably already paginated anyway so it's probably not a huge deal

Reply:
Makes sense, no worries at all. I remember doing the same when I first encountered this in the code.